darts/dataprocessing/transformers/static_covariates_transformer.py

```diff
@@ -301,7 +301,11 @@ def _create_category_mappings(
                     col_map_cat_i = []
                     for cat in categories:
                         col_map_cat_i.append(cat)
-                        inv_col_map_cat[cat] = [col]
+                        if len(categories) > 1:
+                            cat_col_name = str(col) + "_" + str(cat)
+                            inv_col_map_cat[cat_col_name] = [col]
+                        else:
+                            inv_col_map_cat[cat] = [col]
```

Comment:
I think we can always take the cat_col_name. As you said, if there are duplicate categories across different categorical features, we need to avoid appending to the wrong key.
```suggestion
                        cat_col_name = str(col) + "_" + str(cat)
                        inv_col_map_cat[cat_col_name] = [col]
```